multi-agent/run/scripts/migrate_builtin_system.py

```
@@ -0,0 +1,379 @@
+        try:
+            raw = urlsafe_b64decode(value)
+            return len(raw) >= FERNET_MIN_RAW_LENGTH and raw[0] == FERNET_VERSION_BYTE
+        except (binascii.Error, ValueError):
+            return False
+
+    def _decrypt_fields(fields: dict, resource_id: str = "", identity_key: str = "") -> dict:
+        if not isinstance(fields, dict):
+            raise ValueError(
+                f"Overlay {resource_id}/{identity_key} has non-dict fields "
+                f"(got {type(fields).__name__}) — skipping to avoid data corruption"
+            )
```

Ruff (0.16.0)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)